src/EditorFeatures/TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -163,10 +163,13 @@ protected static int CompareRange(LSP.Range r1, LSP.Range r2)
             return compareLine != 0 ? compareLine : compareChar;
         }
 
-        protected static string ApplyTextEdits(LSP.TextEdit[] edits, SourceText originalMarkup)
+        protected static string ApplyTextEdits(LSP.TextEdit[]? edits, SourceText originalMarkup)
         {
+            if (edits is null)
+                return originalMarkup.ToString();
+
             var text = originalMarkup;
-            foreach (var edit in edits)
+            foreach (var edit in edits.OrderByDescending(x => x.Range.Start.Line).ThenByDescending(x => x.Range.Start.Character))
```

Comment:
📝 Might be better to map the `LSP.TextEdit` instances back to `TextChange` and use an existing helper to apply it.

Reply:
➡️ Rewrote this method. Much cleaner.